src/Pages/Dashboard/UploadData.jsx

```diff
@@ -0,0 +1,71 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+import Papa from "papaparse";
+import { apply, compose, lift, splitAt, zipObj } from "ramda";
+
+const timestamp = Date.now();
+
+class UploadData extends React.Component {
+  sendData = data => {
+    this.props.firebase
+      .database()
+      .ref(this.props.sensorId + "/" + timestamp)
+      .set(data);
+  };
+
+  handleFile = event => {
+    const file = event.target.files[0];
+    const keys = this.props.keys;
+    const onDataUploaded = this.props.onDataUploaded;
+    const onError = this.props.onError;
+    const sendData = this.sendData;
+
+    Papa.parse(file, {
+      skipEmptyLines: true,
+      error: function(err, file, inputElem, reason) {
+        onError({ err, file, inputElem, reason });
+      },
+      complete: function(results) {
```

Comment:
I would recommend using `step` to process the data from the csv instead, as it's a callback per line being processed, rather than processing at the end. 

This way, it doesn't freeze the event loop on large files (because we're processing them step by step, rather than all at once at the end).